R/Data-methods.R

```diff
@@ -316,60 +316,63 @@ setMethod(
   }
 )
 
-## --------------------------------------------------
-## Update a DataParts object
-## --------------------------------------------------
+# DataParts-update ----
 
-##' Update method for the "DataParts" class
-##'
-##' Add new data to the \code{\linkS4class{DataParts}} object
-##'
-##' @param object the old \code{\linkS4class{DataParts}} object
-##' @param x the dose level (one level only!)
-##' @param y the DLT vector (0/1 vector), for all patients in this cohort
-##' @param ID the patient IDs
-##' @param \dots not used
-##' @return the new \code{\linkS4class{DataParts}} object
-##'
-##' @example examples/Data-method-update-DataParts.R
-##' @export
-##' @keywords methods
-setMethod("update",
-          signature=
-          signature(object="DataParts"),
-          def=
-          function(object,
-                   x,
-                   y,
-                   ID=(if(length(object@ID)) max(object@ID) else 0L) + seq_along(y),
-                   ...){
+#' Updating `DataParts` Objects
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' A method that updates existing [`DataParts`] object with new data.
+#'
+#' @param object (`DataParts`)\cr object you want to update.
+#' @param check (`flag`)\cr whether the validation of the updated object should be conducted.
+#'   See help for [`update-Data-method`] for more details on the use case of this parameter.
+#' @param ... passed to the first inherited method `update` after this current method,
+#'   i.e. [`update-Data-method`].
+#'
+#' @return The new, updated [`DataParts`] object.
+#'
+#' @aliases update-DataParts-method
+#' @example examples/Data-method-update-DataParts.R
+#' @export
+#'
+setMethod(
+  f = "update",
+  signature = signature(object = "DataParts"),
+  definition = function(object, check = TRUE, ...) {
+    x <- list(...)$x
+    y <- list(...)$y
```

Comment:
I think it is better to have these as explicit arguments (as it was before)

Reply:
Sure, but then I would need to duplicate the help page for this. As I think inheritsParam for a single parameter wont work here....

Reply:
Just these (`x`, `y`) or also `ID` as it used to be? If you want me to add the `ID` too, then I would need to initiate it do default as in `update-Data-method` which means duplicated code. And anyway, we pass other arguments to `update-Data-method` via `...`, for example `new_cohort`

Reply:
ok can you try please? `@inheritParams` could maybe still work, otherwise can copy those argument descriptions. not a big deal. generally the 3 dots thing we need to be careful with

Reply:
the point is arguments we also use in this method should be passed explicitly. other things that only goes upstream can be passed in ...

Reply:
Sure, Tx!